xla/service/gpu/stream_attribute_annotator.h

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2024 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_STREAM_ATTRIBUTE_ANNOTATOR_H_
+#define XLA_SERVICE_GPU_STREAM_ATTRIBUTE_ANNOTATOR_H_
+
+#include "absl/strings/string_view.h"
+#include "xla/hlo/ir/hlo_computation.h"
+#include "xla/hlo/ir/hlo_instructions.h"
+#include "xla/hlo/ir/hlo_module.h"
+#include "xla/service/hlo_pass_interface.h"
+#include "xla/statusor.h"
+
+namespace xla {
+namespace gpu {
+
+// This pass checks to see if:
+//  1.  there's any instruction, that
+//      consumes data from other computes streams,
+//      is missing wait_on_operation_queues attribute.
+//  2.  there's any fusion instruction with non-default
+//      stream fusion root.
+// It will annotate the corresponding instruction with
+// the correct attribute in GpuBackendConfig.
+
+class StreamAttributeAnnotator : public HloModulePass {
```

Comment:
Can you add a bit more documentation that points to async_wrapper pass, because in isolation it's not clear why we need this, and why exactly it's done this way

Reply:
I'm confused by the purpose of wait_on_operation_queues since it was not used by the async op wrapper. A documentation here would be helpful.

Reply:
i added more documentation to the header along with some explanation of wait_on_operation_queues attribute. wait_on_operation_queues is not used in the async wrapper but it's needed for the emitter to emit synchronization thunks since it needs to know the source and target streams